api/src/client/ApiFormationClient.ts

```diff
@@ -161,534 +156,83 @@ export const ApiFormationClient = (config: ApiConfig, logger: LogWriterType): Fo
-      terms: string | undefined
-    ): string | undefined => {
-      return input === "IN_FORM" ? terms : undefined;
-    };
+      formationFormData
+    );
 
-    // eslint-disable-next-line unicorn/consistent-function-scoping
-    const isInFormOrBylaw = (input: "IN_BYLAWS" | "IN_FORM" | undefined): "Herein" | "Bylaw" => {
-      return input === "IN_FORM" ? "Herein" : "Bylaw";
-    };
+    const returnUrl: string = "returnUrl";
+    const foreignGoodStandingFile: InputFile | undefined = undefined;
 
-    const getAdditionalProvisions = (): Provisions => {
-      const additionalProvisions =
-        formationFormData.additionalProvisions?.map((it: string) => {
-          return { Provision: it };
-        }) ?? [];
+    const postBody = makePostBody(userData, returnUrl, config, foreignGoodStandingFile);
+    const logId = logger.GetId();
+    logger.LogInfo(
+      `Formation Health Check - NICUSA - Id:${logId} - Request Sent to ${
+        config.baseUrl
+      }/PrepareFiling data: ${JSON.stringify(postBody)}`
+    );
 
-      switch (toFormationLegalStructure()) {
-        case "limited-partnership": {
-          return {
-            AdditionalLimitedPartnership: {
-              AggregateAmount: formationFormData.combinedInvestment,
-              LimitedCanCreateLimited: formationFormData.canCreateLimitedPartner ? "Yes" : "No",
-              LimitedCanCreateLimitedTerms: formationFormData.createLimitedPartnerTerms,
-              LimitedCanGetDistribution: formationFormData.canGetDistribution ? "Yes" : "No",
-              LimitedCanGetDistributionTerms: formationFormData.getDistributionTerms,
-              LimitedCanMakeDistribution: formationFormData.canMakeDistribution ? "Yes" : "No",
-              LimitedCanMakeDistributionTerms: formationFormData.makeDistributionTerms,
-              GeneralPartnerWithdrawal: formationFormData.withdrawals,
-              DissolutionPlan: formationFormData.dissolution,
-              AdditionalProvisions: additionalProvisions,
-            },
-          };
-        }
-        case "foreign-limited-partnership": {
-          return {
-            AdditionalForeignLimitedPartnership: {
-              AggregateAmount: formationFormData.combinedInvestment,
-              AdditionalProvisions: additionalProvisions,
-            },
-          };
-        }
-        case "limited-liability-company": {
-          return {
-            AdditionalLimitedLiabilityCompany: {
-              OtherProvisions: additionalProvisions,
-            },
-          };
-        }
-        case "limited-liability-partnership": {
-          return {
-            AdditionalLimitedLiabilityPartnership: {
-              OtherProvisions: additionalProvisions,
-            },
-          };
-        }
-        case "nonprofit":
-        case "veteran-nonprofit": {
+    return axios
+      .post(`${config.baseUrl}/PrepareFiling`, postBody, {
+        headers: {
+          "Content-Type": "application/json",
```

Comment:
[boulder] Currently, the Formation API can support application/json and text/plain, but long term they're going to exclusively use text/plain. Our goal is to only use text/plain. Can we account for this here?

Reply:
done changed to  text/plain